system-helper/flatpak-system-helper.c

```diff
@@ -25,21 +25,29 @@
 #include <string.h>
 #include <gio/gio.h>
 #include <polkit/polkit.h>
+#include <sys/mount.h>
+#include <gio/gunixmounts.h>
 
 #include "flatpak-dbus-generated.h"
 #include "flatpak-dir-private.h"
 #include "flatpak-oci-registry-private.h"
 #include "flatpak-error.h"
+#include "flatpak-utils-private.h"
 
 static PolkitAuthority *authority = NULL;
 static FlatpakSystemHelper *helper = NULL;
 static GMainLoop *main_loop = NULL;
 static guint name_owner_id = 0;
 
+G_LOCK_DEFINE (ongoing_pulls);
+static GPtrArray *ongoing_pulls = NULL;
+
 static gboolean on_session_bus = FALSE;
 static gboolean no_idle_exit = FALSE;
 
 #define IDLE_TIMEOUT_SECS 10 * 60
+#define MAX_PULLS_PER_USER 16
+#define MOUNT_TIMEOUT 15
```

Comment:
Following the idle timeout example, this should probably be MOUNT_TIMEOUT_SECS ?